sql/expression/function/days_test.go

```diff
@@ -264,3 +264,106 @@ func TestFromDays(t *testing.T) {
 		})
 	}
 }
+
+func TestLastDay(t *testing.T) {
+	tests := []struct {
+		arg  sql.Expression
+		exp  interface{}
+		err  bool
+		skip bool
+	}{
+		{
+			arg: expression.NewLiteral(nil, types.Null),
+			exp: nil,
+		},
+		{
+			arg: expression.NewLiteral("notadate", types.Text),
+			exp: nil,
+		},
+		{
+			arg: expression.NewLiteral(-10, types.Int32),
+			exp: nil,
+		},
+		{
+			arg: expression.NewLiteral(366.13, types.Float32),
+			exp: nil,
+		},
+		{
+			arg: expression.NewLiteral("0004-02-01", types.Text),
+			exp: time.Date(4, 2, 29, 0, 0, 0, 0, time.UTC),
+		},
+		{
+			skip: true, // we should parse 00 days
+			arg:  expression.NewLiteral("0001-11-00", types.Text),
+			exp:  time.Date(1, 11, 30, 0, 0, 0, 0, time.UTC),
+		},
+		{
```

Comment:
(minor) might be worth adding another date that has an incorrect day for the month, like `2020-02-30` or something.